File: Maximum sum of non-adjacent elements/Recursion.py
def useRecursion(size,nums):
    if size == 0: return nums[0]
    if size < 0: return 0

    pick = nums[size] +  useRecursion(size-2,nums)
    nonPick = useRecursion(size-1,nums)

    return max(pick,nonPick)

def findMaximum(arr):
    size =  len(arr)-1

    return useRecursion(size,arr)

if __name__ == '__main__':
    print(findMaximum([2,1,4,9]))

File: Maximum sum of non-adjacent elements/Recursion.java
public class Recursion {

    public static int MaximumSum(int[] nums,int size) {
        if(size == 0) return nums[size];
        if(size < 0) return 0;

        int pick =  nums[size] + MaximumSum(nums, size-2);
        int nonPick = MaximumSum(nums, size-1);

        return Math.max(pick,nonPick);
    }
    public static void main(String[] args) {
        int[] nums = {2 ,1 ,4 ,9};
        int size = nums.length;

        System.out.println("Maximum sum without Adjacent : "+ MaximumSum(nums,size-1));
    }
}